server/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java

```diff
@@ -2666,4 +2666,401 @@ public void test_can_resolve_index_through_aliased_relation() throws Exception {
             TestingHelpers.isSQL("MATCH((t.body_ft NULL), 'foo') USING best_fields WITH ({})")
         );
     }
+
+    @Test
+    public void testSubscriptExpressionWithUnknownRootWithSessionSetting() throws Exception {
+        /*
+         * create table tbl (a int);
+         * select unknown['u'] from tbl; --> ColumnUnknownException
+         * set errorOnUnknownObjectKey = false;
+         * select unknown['u'] from tbl; --> ColumnUnknownException
+         */
+        var executor = SQLExecutor.builder(clusterService)
+            .addTable("CREATE TABLE tbl (a int)")
```

Comment:
test cases are verbose, please let me know if this is too much. I wanted to clearly show the comparisons with the session setting set to true and false. Also to document the current bahaviours that are tricky and possibly bugs.